File: backend/src/main/java/com/yourwatchrental/watchrental/rental/dto/request/RentalFilterRequestDTO.java
package com.yourwatchrental.watchrental.rental.dto.request;

import com.yourwatchrental.watchrental.rental.PaymentMethod;
import com.yourwatchrental.watchrental.rental.PaymentStatus;
import com.yourwatchrental.watchrental.rental.RentalStatus;

import java.time.LocalDate;
import java.util.UUID;

public record RentalFilterRequestDTO(
        RentalStatus rentalStatus,
        PaymentStatus paymentStatus,
        PaymentMethod paymentMethod,
        UUID userId,
        UUID watchId,
        UUID branchId,
        LocalDate startDateFrom,
        LocalDate startDateTo,
        LocalDate endDateFrom,
        LocalDate endDateTo
) {}

File: frontend/src/app/core/models/rentals/rental-filter-request.dto.ts
import { PaymentMethod } from './payment-method';
import { PaymentStatus } from './payment-status';
import { RentalStatus } from './rental-status';

export interface RentalFilterRequestDTO {
  rentalStatus?: RentalStatus | null;
  paymentStatus?: PaymentStatus | null;
  paymentMethod?: PaymentMethod | null;
  userId?: string | null;
  watchId?: string | null;
  branchId?: string | null;
  startDateFrom?: string | null;
  startDateTo?: string | null;
  endDateFrom?: string | null;
  endDateTo?: string | null;
}